subprojects/dependency-management/src/main/java/org/gradle/api/internal/catalog/DefaultVersionCatalogBuilder.java

```diff
@@ -177,40 +178,31 @@ public void withContext(String context, Runnable action) {
         }
     }
 
-    private RuntimeException throwVersionCatalogProblemException(ProblemBuilder problem) {
-        throw throwErrorWithNewProblemsApi("Invalid catalog definition", ImmutableList.of(problem.build()));
-    }
-
-    @Nonnull
-    public ProblemBuilder createVersionCatalogError(String message, VersionCatalogProblemId catalogProblemId) {
-        return getProblemService().createProblemBuilder()
-            .label(message)
-            .documentedAt(userManual(VERSION_CATALOG_PROBLEMS, catalogProblemId.name().toLowerCase()))
-            .noLocation()
-            .type(catalogProblemId.name())
-            .group(VERSION_CATALOG_ID)
-            .severity(ERROR);
+    private RuntimeException throwVersionCatalogProblemException(ReportableProblem problem) {
```

Comment:
💭 If `createVersionCatalogError` was made `static`, perhaps this should also be made `static`?